### ISSUE TITLE
Animation speed (original game issue, really)

### ISSUE THREAD

**Issue by Digital-Monk** [afritz1/OpenTESArena#190]
This is a VERY low priority request.

I've always found Arena's animations (NPC idling, flames, water fountains, etc) to have a Keystone Cop feel to them -- just too fast, like everybody is coked out of their mind.

I was fiddling about today with Misc / Time Scale (thank you for that feature) and noticed that 0.75 feels much more natural to me for NPC idling animations, flames, fountains, and that sort of thing.  0.50 felt more natural when watching people walk around town, but I was in a city in Black Marsh, so I expected more New Orleans strolling than New York hubbub.  0.50 also started to feel a little annoying because it was impacting my movement -- I didn't really notice at 0.75.

I guess I was wondering how hard it would be to allow tuning of these non-critical things separate from the game's overall timescale (or in addition to it).  So, Time Scale would affect combat speed and everything that "mattered", but there could be a separate (or additional) scale factor for animations and NPC movement (things that don't really matter, but do affect immersion).

If it's a headache, I'm perfectly happy to run somewhere in the 0.50 - 0.75 range on global Time Scale.  I'm not really a twitch player anyway.

And if you haven't tried slowing things down, it's worth it just for a leisurely stroll around town and some shopping ;)

**Comment by afritz1** [comment]
I was thinking of maybe removing the TimeScale feature at some point because it felt like OpenTESArena shouldn't have any kind of dependency/relation to DOSBox; it should be its own thing. Tweaking enemy speed, citizen speed, and whatnot could probably be in a text file someday.

**Comment by hfc2x** [comment]
Hey, I just wanted to chime-in, mostly because the gameplay speed was an intentional "feature" in the original game.

Arena was so ~~unoptimized~~ demanding to run for CPUs at the time, that Bethesda intentionally made attack animations and also movement to go at very high speeds, so the game would play at a sane pace due to the permanent slowdown.

If you play the game in a more modern computer with a better CPU than intended, or through DOSBox with the max cycles setting, the game will play at absurdly silly speeds as a consequence. For this reason, people nowadays tend to believe that Arena _was_, in fact, intended to be stupidly fast and demand really high reflexes from the player, when it was never the intention.

So really, if there is to be a user-adjustable setting that controls certain instances of game speed, the speed comparable to that which you get on DOSBox with max cycles should instead be called "turbo speed" or something of the sort, rather than being "1.00 speed", since it's an unintended consequence of the original's design. The speed that's considered the "target", is the one you get by forcing DOSBox to run the game at a locked maximum of between 20.000 and 30.000 cycles inclusive.

I apologize for using DOSBox as a metric here, since you explicitly stated you don't want OpenTESArena to have any relation to it, but I thought it was the more "universal" metric to use for a comparison.

**Comment by Thunderforge** [comment]
For completeness: I'll mention that Daggerfall did have a speed slider, but it also affected gameplay; the player gained experience more slowly at lower speeds (requiring slower reflexes) than at higher speeds (requiring faster reflexes). 

I agree that a speed slider would be helpful to include in OpenTESArena, and I'm ambivalent to whether or not it behaves as Daggerfall did or if it has no mechanical impact on gameplay.

**Comment by afritz1** [comment]
Both good points.

Entity animations would probably have an extra multiplier paired with the delta time to tweak these kinds of things on a case by case basis.

**Comment by Allofich** [comment]
The French box (https://www.mobygames.com/game/dos/elder-scrolls-arena/cover-art/gameCoverId,1357/) says a 386/25 Mhz minimum, 386/33 Mhz recommended.

The official strategy guide book "Codex Scientia" says the same, and also has a note saying Intel 486/33+ "really cooks!", which I take to mean is even better.

I've been curious about how Arena would run on a real system of the time for a while now, as I'd only ever played it through DOSBox-X. I noticed, in addition to the animation speed of NPCs, the walking speed of NPCs was also fast if you played with a high cycles count. Also the speed of the flashing red outline on locations selected on the world map seemed too fast in DOSBox-X.

Although I don't have real hardware to test with, I just tested through 86Box to hopefully more accurate results than DOSBox-X (which itself aims to be more accurate than regular DOSBox). Running the game with an emulated 386/33 Mhz, the framerate runs at a crawl by modern standards, but I'm sure it would have been more acceptable at the time. At this lower framerate, though, the movement speed of NPCs is more realistic. Animations of non-moving NPCs are also of course more normal looking. As for the flashing red outline, it's also slower than in DOSBox-X (even if you approximating a 386/33 Mhz with cycles there).

Upping the speed to 486/33 Mhz makes the framerate nicer, but the NPC animations and movement are already starting to look a bit fast.

The slowest time scale (0.50) in OpenTESArena looks similar to the 486/33 speed in 86Box.

**Comment by afritz1** [comment]
Good to know. I did some informal testing of camera rotation speed in OpenTESArena. At 1.0 time scale, it's about 7 1/2 full spins in 10 seconds, and at 0.5 time scale about 3 1/2 full spins. When implementing it, I tried to match it to Arena in DOSBox, so hopefully it's not too far off from what the original game was like.

**Comment by Allofich** [comment]
I don't know how turn speed relates to animation speed, etc. but in 86Box with an emulated 486/33 I get about 2 1/2 full spins in 10 seconds. With an emulated 386/33 I get about 1 1/4 full spins.

Something to consider is ease/comfort of use vs. game difficulty. I would think faster camera rotation speed, like OpenTESArena has now, feels more modern and comfortable to use, but it would also make the game easier than the original, such as when you're attacked from the side or back.

Of course, when it comes to gameplay, difficulty, etc. there are all kinds of considerations, like whether or not you want better AI pathfinding. The enemies in the original game seem to pretty much just walk in a straight line toward you and easily get caught on doorways, etc., so if they didn't do that anymore because of smarter pathfinding the game would probably be harder.

I just wanted to mention that giving the player faster turning will probably make the game easier than the original, so that's something to keep in mind.

**Comment by afritz1** [comment]
Yup, I just mentioned camera rotation since it's also affected by overall game speed in DOSBox. Only 2 1/2 spins seems very, very slow, maybe enough for the RPG people of the time though. If I remember correctly, Terminator: Rampage runs around a fixed 12 fps or so and therefore appears very choppy but doesn't feel like slow motion.

**Comment by Kirtai** [comment]
Shouldn't the camera speed be adjustable as an improvement?

**Comment by afritz1** [comment]
Horizontal and vertical camera sensitivity are in the options. Don't remember if they affect classic mode though.

**Comment by Chatplosion** [comment]
How about taking out some real legacy hardware and test the real FPS? Or get in touch with the game developers for information.
I do believe that there is a speed limitation on hardware.

**Comment by XJDHDR** [comment]
> If you do add different options, I think one of them should be clearly labeled as being the settings for the original game, and it should either be the default, or the user should be prompted to choose between the "original behavior" or a more "modern behavior". 

I think the biggest problem with your suggestion is that AFAIK, no one has been able to determine what the "original behaviour" is.
